denoise_tests.py

```diff
@@ -0,0 +1,385 @@
+import argparse
+import logging
+import os
+import sys
+from timeit import default_timer as timer
+
+import cv2 as cv
+import bm3d
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import psutil
+import skimage
+import tifffile
+import zarr
+from scipy import fftpack
+from skimage.exposure import rescale_intensity
+from skimage.restoration import (
+    estimate_sigma,
+    denoise_nl_means,
+    denoise_bilateral,
+    denoise_tv_chambolle,
+    denoise_wavelet
+)
+
+import compress_zarr
+
+try:
+    import imagej
+    import scyjava
+    # initialize Fiji with the external non-local means plugin
+    IMAGEJ = imagej.init(['sc.fiji:fiji', 'com.github.thorstenwagner:ij-nl-means'], headless=False)
+    IMAGEJ.ui().showUI()
+except ImportError:
+    IMAGEJ = None
+
+
+def identity(data):
+    return data
+
+
+def run_ij_plugin(arr, plugin, args):
```

Comment:
Not a comment on the code, but general question.  Does that mean we would be able to run any ImageJ/Fiji plugins without the UI?

Reply:
It depends, ImageJ2/Fiji plugins can (I think for the most part) run headless, as long as they don't instantiate any Java-based GUI components. Certain ImageJ1 plugins that make use of  the `WindowManager` to get the activate image, or display results in tables/plots will probably not work. There is also the pain of converting between ImageJ1 data structures (`ImagePlus`) and ImageJ2 data structures (imglib2-based `RandomAccessibleInterval`) when running ImgaeJ1 plugins from `pyimagej`, which is based on ImageJ2.

More info here https://imagej.net/learn/headless